evaluation-tasks/util/s3.py

```diff
@@ -0,0 +1,153 @@
+"""
+Utility functions for interacting with an S3 bucket.
+"""
+
+import os
+import subprocess
+
+import boto3
+from luigi.contrib.s3 import S3Client
+from botocore.client import ClientError
+import luigi
+
+from .luigi import WorkTask
+
+
+class CacheTarCorpus(WorkTask):
+    """
+    If the tar file is cached in S3, we simply retrieve it and untar
+    it, and skip the pipeline.
+
+    If the tar file is NOT in S3, we run the pipeline, create the
+    tar-file, and upload it to S3.
+    """
+
+    task_name = luigi.Parameter()
+    bucket = luigi.Parameter()
+    region = luigi.Parameter()
+    next_task = luigi.Parameter()
+    # TODO: How do we support params for next_task??
+
+    def requires(self):
+        return EnsureBucket(self.bucket, self.region)
+
+    @property
+    def name(self):
+        return type(self).__name__
+
+    def run(self):
+        tarfile = f"{self.task_name}.tar.gz"
+        pathtarfile = f"{self.workdir}/{tarfile}"
+        client = S3Client()
+        s3cache = os.path.join(f"s3://{self.bucket}/", tarfile)
+        if client.exists(s3cache):
+            # TODO: Create a new task to untar this a move it out of
+            # workdir into the main evaluation dir so it is good to go.
```

Comment:
If the tar exists on s3 we should untar it and move it into main dir after download. That way this will behave the same as finalize corpus